app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -638,38 +651,65 @@ private void scanPokemon(Bitmap pokemonImage, String filePath) {
         estimatedPokemonLevel= getPokemonLevel(pokemonImage);
         Log.d("Estimated pokelevel: " + estimatedPokemonLevel, "nahojjjen debug ghastly crash");
 
-        Bitmap name = Bitmap.createBitmap(pokemonImage, displayMetrics.widthPixels / 4, (int) Math.round(displayMetrics.heightPixels / 2.22608696), (int) Math.round(displayMetrics.widthPixels / 2.057), (int) Math.round(displayMetrics.heightPixels / 18.2857143));
-        name = replaceColors(name, 68, 105, 108, Color.WHITE, 200);
-        tesseract.setImage(name);
-        //System.out.println(tesseract.getUTF8Text());
-        pokemonName = tesseract.getUTF8Text().replace(" ", "").replace("1", "l").replace("0", "o").replace("Sparky", getString(R.string.pokemon133)).replace("Rainer", getString(R.string.pokemon133)).replace("Pyro", getString(R.string.pokemon133));
-
-        if (pokemonName.toLowerCase().contains("nidora")){
-            boolean isFemale = isNidoranFemale(pokemonImage);
-            if(isFemale){
-                pokemonName = getResources().getString(R.string.pokemon029);
-            }else{
-                pokemonName = getResources().getString(R.string.pokemon032);
+        Bitmap name = Bitmap.createBitmap(pokemonImage, displayMetrics.widthPixels / 4,
+                                            (int) Math.round(displayMetrics.heightPixels / 2.22608696),
+                                            (int) Math.round(displayMetrics.widthPixels / 2.057),
+                                            (int) Math.round(displayMetrics.heightPixels / 18.2857143));
+        String hash = "name" + hashBitmap(name);
+        pokemonName = OCRCache.get(hash);
+        if (pokemonName == null) {
+            name = replaceColors(name, 68, 105, 108, Color.WHITE, 200);
```

Comment:
> I'd still not bother doing this for OCR caching though because this is still a lot of work to save a few bytes of OCR cache.

Fine with that, with all those multiplications this is likely more expensive than the hashing.